terraform/account-wide-infrastructure/modules/glue/iam.tf

```diff
@@ -0,0 +1,68 @@
+resource "aws_iam_role" "glue_service_role" {
+  name = "glue_service_role"
+
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Service" : "glue.amazonaws.com"
+        },
+        "Action" : "sts:AssumeRole"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "glue_service_role_policy" {
+  name = "glue_service_role_policy"
+  role = aws_iam_role.glue_service_role.name
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      # {
+      #   "Effect" : "Allow",
+      #   "Action" : [
+      #     "glue:*",
+      #     "s3:GetBucketLocation",
+      #     "s3:ListBucket",
+      #     "s3:ListAllMyBuckets",
+      #     "s3:GetBucketAcl",
+      #     "cloudwatch:PutMetricData"
+      #   ],
+      #   "Resource" : ["*"]
+      # },
+      {
+        "Effect" : "Allow",
+        "Action" : ["s3:CreateBucket"],
+        "Resource" : ["arn:aws:s3:::aws-glue-*"]
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : ["s3:GetObject", "s3:PutObject", "s3:DeleteObject"],
+        "Resource" : [
+          "arn:aws:s3:::*/*",
+          "arn:aws:s3:::*/*aws-glue-*/*"
+        ]
+      },
+      {
+        "Effect" : "Allow",
+        "Action" : ["s3:GetObject"],
+        "Resource" : [
+          "arn:aws:s3:::crawler-public*",
```

Comment:
Its part of the default policy provided by amazon - AWSGlueServiceRole. I chopped it down a little but it might be simpler just to invoke the whole thing (change below)

`resource "aws_iam_role_policy_attachment" "glue_service" {
  role       = aws_iam_role.glue_service_role.id
  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSGlueServiceRole"
}`